Unit test expects SDL version to stay 2.0.x forever

**What doesn't work?**

`sdl2/test/version_test.py` has:

```
def test_SDL_GetVersion():
    v = sdl2.SDL_version()
    sdl2.SDL_GetVersion(ctypes.byref(v))
    assert type(v) == sdl2.SDL_version
    assert v.major == 2
    assert v.minor == 0
    assert v.patch >= 5
```

This assumes that all SDL versions compatible with SDL 2.0.0 will be versioned 2.0.x.

However, the stable version after SDL 2.0.22 is going to be 2.24.0, with prereleases identifying themselves as 2.23.x. See [the documentation](https://github.com/libsdl-org/SDL/blob/main/docs/README-versions.md) for details.

The motivation for this change was that in the previous numbering scheme, the `.0.` was essentially useless (if it only increases on incompatible changes, then it's redundant with the major version), and there was no way to signal bugfix-only releases to fix regressions or other serious bugs without adding new features.

Next time SDL breaks compatibility, the new/incompatible version will call itself SDL 3, with filenames likely to look like `libSDL3.so.0` or `SDL3.dll`.

**How To Reproduce**

Upgrade SDL2 to a git snapshot or a future stable release, then run py-sdl2's unit tests. You can see this happen in Debian's CI system when it tries to run tests against a `libsdl2` git snapshot from `experimental`, as listed in <https://ci.debian.net/packages/p/pysdl2/unstable/amd64/>:

```
self = <sdl2.test.version_test.TestSDLVersion object at 0x7fc0ba1caac0>

    def test_SDL_GetVersion(self):
        v = version.SDL_version()
        version.SDL_GetVersion(ctypes.byref(v))
        assert type(v) == version.SDL_version
        assert v.major == 2
>       assert v.minor == 0
E       assert 23 == 0
E         +23
E         -0
```

**Platform (if relevant):**
 - OS: any, tested on Debian `unstable` rolling release with a SDL git snapshot from `experimental`
 - Python Version: any, tested with 3.10.4
 - SDL2 Version: recent git snapshots (version is reported as 2.23.0 for now, a 2.23.1 development prerelease is expected in future)
 - Using pysdl2-dll: I don't think so?

--- a/doc/news.rst
+++ b/doc/news.rst
@@ -22,6 +22,7 @@ New Features:
 * Added an informative warning for users using pysdl2-dll with PySDL2 on an
   Apple Silicon Mac.
 
+
 Fixed Bugs:
 
 * Fixed a bug in :func:`~sdl2.rw_from_object` where calling 
@@ -46,6 +47,8 @@ Fixed Bugs:
   the SDL2 headers.
 * Fixed export of ``SDL_TIMER_RESOLUTION`` hint (was previously not accessable
   through the ``sdl2`` namespace).
+* Updated ``sdl2.dll.version`` to better handle SDL2's new versioning format
+  and fixed unit tests accordingly (issue #228).
 
 
 0.9.11

--- a/sdl2/dll.py
+++ b/sdl2/dll.py
@@ -63,12 +63,23 @@ class SDL_version(Structure):
     ]
 
 def _version_str_to_int(s):
+    # Convert an SDL version string to an integer (e.g. "2.0.18" to 2018)
     v = [int(n) for n in s.split('.')]
-    return v[0] * 1000 + v[1] * 100 + v[2]
+    if v[1] > 0:
+        # For SDL2 >= 2.23.0 (new version scheme): 2.23.0 -> 2230
+        return v[0] * 1000 + v[1] * 10 + v[2]
+    else:
+        # For SDL2 <= 2.0.22 (old version scheme): 2.0.22 -> 2022
+        return v[0] * 1000 + v[1] * 100 + v[2]
 
 def _version_int_to_str(i):
     v = str(i)
-    v = [v[0], v[1], str(int(v[2:4]))]
+    if int(v[1]) > 0:
+        # For SDL2 >= 2.23.0 (new version scheme): 2230 -> 2.23.0
+        v = [v[0], str(int(v[1:3])), v[3]]
+    else:
+        # For SDL2 <= 2.0.22 (old version scheme): 2022 -> 2.0.22
+        v = [v[0], v[1], str(int(v[2:4]))]
     return ".".join(v)
 
 def _so_version_num(libname):
@@ -322,7 +333,7 @@ class DLL(object):
     @property
     def version(self):
         """int: The version of the loaded library in the form of a 4-digit
-        integer (e.g. '2008' for SDL 2.0.8, '2010' for SDL 2.0.10).
+        integer (e.g. '2008' for SDL 2.0.8, '2231' for SDL 2.23.1).
         """
         return self._version
 

--- a/sdl2/test/sdlimage_test.py
+++ b/sdl2/test/sdlimage_test.py
@@ -90,7 +90,7 @@ def test_IMG_Linked_Version():
     v = sdlimage.IMG_Linked_Version()
     assert isinstance(v.contents, version.SDL_version)
     assert v.contents.major == 2
-    assert v.contents.minor == 0
+    assert v.contents.minor >= 0
     assert v.contents.patch >= 1
 
 def test_IMG_Init():

--- a/sdl2/test/sdlmixer_test.py
+++ b/sdl2/test/sdlmixer_test.py
@@ -19,7 +19,7 @@ def test_Mix_Linked_Version():
     v = sdlmixer.Mix_Linked_Version()
     assert isinstance(v.contents, version.SDL_version)
     assert v.contents.major == 2
-    assert v.contents.minor == 0
+    assert v.contents.minor >= 0
     assert v.contents.patch >= 0
 
 @pytest.mark.skipif(sdlmixer.dll.version < 2004, reason="Broken in official binaries")

--- a/sdl2/test/sdlttf_test.py
+++ b/sdl2/test/sdlttf_test.py
@@ -66,7 +66,7 @@ def test_TTF_Linked_Version(with_sdl_ttf):
     v = sdlttf.TTF_Linked_Version()
     assert isinstance(v.contents, version.SDL_version)
     assert v.contents.major == 2
-    assert v.contents.minor == 0
+    assert v.contents.minor >= 0
     assert v.contents.patch >= 12
 
 @pytest.mark.skipif(sdlttf.dll.version < 2018, reason="not available")

--- a/sdl2/test/version_test.py
+++ b/sdl2/test/version_test.py
@@ -16,7 +16,7 @@ def test_SDL_GetVersion():
     sdl2.SDL_GetVersion(ctypes.byref(v))
     assert type(v) == sdl2.SDL_version
     assert v.major == 2
-    assert v.minor == 0
+    assert v.minor >= 0
     assert v.patch >= 5
 
 def test_SDL_VERSIONNUM():

